### PR TITLE
Add flash when activating the osu! logo (and other UI improvements)

### PR DESCRIPTION
**Pull request by peppy**
Just some minor stuff I noticed missing.

### CODE REVIEW

File: osu.Game/Graphics/OsuColour.cs [hunk 1]
```diff
@@ -24,8 +24,13 @@ private static Color4 FromHex(string hex)
                 255);
         }
 
-        // See https://github.com/ppy/osu-web/blob/master/resources/assets/less/colors.less
+        //used for osu! logo and triangles contained within.
+        public Color4 OsuPink = FromHex(@"e967a1");
+        public Color4 OsuPinkLight = FromHex(@"ff7db7");
+        public Color4 OsuPinkDark= FromHex(@"de5b95");
```
Comment (SirCmpwn) [inline osu.Game/Graphics/OsuColour.cs]: If it's only used in one place it should just hang out in that place.

Comment (peppy) [inline osu.Game/Graphics/OsuColour.cs]: It's a very important colour so I'd rather have it in the palette.

Comment (SirCmpwn) [inline osu.Game/Graphics/OsuColour.cs]: It's confusing next to Pink, then. Why do we have two pinks?

File: osu.Game/Screens/Menu/OsuLogo.cs [hunk 2]
```diff
@@ -62,6 +65,8 @@ public bool Ripple
         }
 
         public bool Interactive = true;
+        private Box colourLayer;
```
Comment (Tom94) [inline osu.Game/Screens/Menu/OsuLogo.cs]: It doesn't look like this is ever used.

File: osu.Game/Screens/Menu/OsuLogo.cs [hunk 3]
```diff
@@ -225,8 +244,8 @@ private Color4 getTriangleShade()
             {
```
Comment (Tom94) [inline osu.Game/Screens/Menu/OsuLogo.cs]: Just a comment on the side: We may want to drop transparent triangles within the regular triangles background as well. It looks to me like flyte's designs use opaque triangles in all samples (and I personally prefer their look).

Comment (peppy) [inline osu.Game/Screens/Menu/OsuLogo.cs]: agreed

**Comment by peppy** [comment]
Now depends on https://github.com/ppy/osu-framework/pull/419

File: osu.Game/Beatmaps/Drawables/BeatmapPanel.cs [hunk 4]
```diff
@@ -65,15 +69,11 @@ public BeatmapPanel(BeatmapInfo beatmap)
...
+                triangles = new Triangles
                 {
-                    // The border is drawn in the shader of the children. Being additive, triangles would over-emphasize
-                    // the border wherever they cross it, and thus they get their own masking container without a border.
-                    Masking = true,
-                    CornerRadius = Content.CornerRadius,
                     RelativeSizeAxes = Axes.Both,
-                    BlendingMode = BlendingMode.Additive,
-                    Colour = new Color4(20, 43, 51, 255),
+                    ColourLight = OsuColour.FromHex(@"3a7285"),
```
Comment (Tom94) [inline osu.Game/Beatmaps/Drawables/BeatmapPanel.cs]: I wonder if it's feasible to grab colours from OsuColour, and then apply some hypothetical darkening or brightening extension methods. Would be great semantically, and would mean robustness w.r.t. palette.

File: osu.Game/Graphics/Backgrounds/Triangles.cs [hunk 5]
```diff
@@ -65,12 +64,14 @@ protected virtual Triangle CreateTriangle()
                 RelativePositionAxes = Axes.Both,
                 Scale = new Vector2(scale),
                 // Scaling height by 0.866 results in equiangular triangles (== 60° and equal side length)
+                Colour = GetTriangleShade(),
```
Comment (Tom94) [inline osu.Game/Graphics/Backgrounds/Triangles.cs]: You should leave the comment above the correct line.

File: osu.Game/Beatmaps/Drawables/BeatmapPanel.cs (hunk 4 above)
Comment (peppy) [inline osu.Game/Beatmaps/Drawables/BeatmapPanel.cs]: yeah, quite possibly. i haven't yet figured out if flyte's triangle ranges are consistent across elements though (i tend to think they aren't atm).

File: osu.Game/Graphics/UserInterface/OsuTextBox.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colour)
+        {
+            BorderColour = colour.Yellow;
+        }
+
+        protected override bool OnFocus(InputState state)
+        {
+            BorderThickness = 3;
```
Comment (Tom94) [inline osu.Game/Graphics/UserInterface/OsuTextBox.cs]: I really like this usage of BorderThickness 👍

File: osu.Game/Graphics/Backgrounds/Triangles.cs (hunk 5 above)
Comment (peppy) [inline osu.Game/Graphics/Backgrounds/Triangles.cs]: i think it still is. maybe a github display bug?

File: osu.Game/Graphics/UserInterface/OsuTextBox.cs (hunk 6 above)
Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuTextBox.cs]: we will need to make a transform for this in the future (it should fade eventually).

File: osu.Game/Graphics/UserInterface/OsuTextBox.cs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            BorderThickness = 0;
+
+            base.OnFocusLost(state);
+        }
+    }
+
+    public class OsuPasswordTextBox : OsuTextBox
+    {
+        protected virtual char MaskCharacter => '*';
```
Comment (Tom94) [inline osu.Game/Graphics/UserInterface/OsuTextBox.cs]: Any plan to replace this with a centered and bold dot unicode character?

Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuTextBox.cs]: quite possible, yes. will need to alter the font files before doing so.

**Review by Tom94 — APPROVED** [review]

**Pull request merged**